### PR TITLE
Update vignettes

### PR DESCRIPTION
**Pull request by agila5**
First draft for the first half. I hope I will finish it in a couple of days

### CODE REVIEW

**Comment by agila5** [comment]
Hi @Robinlovelace. I modified several things with this PR, nothing vital for the package. Still, I changed several paragraphs in the introductory vignette (please note that the introductory vignette is not shown by default in the "Files changed" since I modified many things), docs, comments, and help pages. I also added a new vignette explaining the differences between the tessellation used by different providers. 

As always, feel free to ask any questions.

**Comment by Robinlovelace** [comment]
Great work @agila5, I plan to take a look tomorrow morning. Most likely will make comments, merge and make a follow-up PR or make any obvious improvements / tidying changes directly. Any questions, will still be good to discuss here. Thanks! 

BTW I was thinking a description of how the package works in terms of when vectortranslate is triggered could be good if not already in there - worth a separate vignette or would that fit in the existing ones? Can open a separate issue if :+1: from you.

**Comment by agila5** [comment]
> Great work @agila5, I plan to take a look tomorrow morning. Most likely will make comments, merge and make a follow-up PR or make any obvious improvements / tidying changes directly. Any questions, will still be good to discuss here. Thanks!

Thanks! 

> BTW I was thinking a description of how the package works in terms of when vectortranslate is triggered could be good if not already in there - worth a separate vignette or would that fit in the existing ones? Can open a separate issue if 👍 from you.

I tried to explain that step [here](https://github.com/ITSLeeds/osmextract/blob/66e31c4b56b48f18f84cd135fa97cfae38f54818/R/vectortranslate.R#L47-L52) and [here](https://github.com/ITSLeeds/osmextract/blob/66e31c4b56b48f18f84cd135fa97cfae38f54818/vignettes/osmextract.Rmd#L350-L355). If you think that it's still not clear I can improve the docs (but I wouldn't create a new vignette just for that topic)

**Comment by Robinlovelace** [comment]
> I tried to explain that step [here](https://github.com/ITSLeeds/osmextract/blob/66e31c4b56b48f18f84cd135fa97cfae38f54818/R/vectortranslate.R#L47-L52) and [here](https://github.com/ITSLeeds/osmextract/blob/66e31c4b56b48f18f84cd135fa97cfae38f54818/vignettes/osmextract.Rmd#L350-L355). If you think that it's still not clear I can improve the docs (but I wouldn't create a new vignette just for that topic)

Looks good to me, think it may be worthy of a vignette of its own, partly because that vignette is already quite long, but that's definitely not a priority so will leave as is.

File: R/data.R [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#' A `data.frame` of geographical zones taken from Geofabrik
+#' An `sf` object of geographical zones taken from Geofabrik
```
Comment (Robinlovelace) [inline R/data.R:1]: :+1: more precise + specific.

File: R/download.R [hunk 2]
```diff
@@ -114,6 +114,7 @@ oe_download = function(
       !is.na(file_size) &&
       file_size >= max_file_size
     ) {
+      message("You are trying to download a file from ", file_url)
```
Comment (Robinlovelace) [inline R/download.R:117]: Great the user will see the url +1. I assume they will only see this when the quiet setting is off.

File: R/get.R [hunk 3]
```diff
@@ -58,9 +58,10 @@
 #'   in the resulting data frame. See documentation at
 #'   [gdal.org](https://gdal.org/drivers/vector/osm.html). Check Details in
 #'   [oe_vectortranslate()].
-#' @param extra_tags Which addition columns, corresponding to OSM tags,
-#'   should be in the resulting dataset? `FALSE` by default. Check Details at
-#'   [oe_vectortranslate()] and [oe_get_keys()] .
+#' @param extra_tags Which addition columns, corresponding to OSM tags, should
```
Comment (Robinlovelace) [inline R/get.R]: Should be

```
 #' @param extra_tags Which additional columns, corresponding to OSM tags, should 
```

File: R/match.R [hunk 4]
```diff
@@ -117,7 +117,7 @@ oe_match.sfc_POINT = function(
 
   # Spatial subset according to sf::st_intersects (maybe add a parameter for
   # that)
-  matched_zones = provider_data[place, ]
+  matched_zones = provider_data[place, op = sf::st_intersects]
```
Comment (Robinlovelace) [inline R/match.R:120]: Wouldn't `st_within` be more appropriate?

File: vignettes/osmextract.Rmd [hunk 5]
```diff
@@ -28,25 +28,25 @@ There are important legal considerations that you should be aware of before usin
...
+- **Share-Alike**: If you publicly use any adapted version of this database, or works produced from an adapted database, you must also offer that adapted database under the ODbL.
+- **Keep open**: If you redistribute the database or an adapted version of it, then you may use technological measures that restrict the work (such as DRM) as long as you also redistribute a version without such measures.
 
 In short, publicly using OSM data without attribution or selling datasets derived from it is illegal.
 See the [License/Use Cases page on the OSM wiki](https://wiki.openstreetmap.org/wiki/License/Use_Cases) for detailed use cases.
 
 # Main package functions
 
-The packages is composed of the following main functions: 
+The package is composed of the following main functions:
```
Comment (Robinlovelace) [inline vignettes/osmextract.Rmd:42]: :+1:

File: vignettes/osmextract.Rmd [hunk 6]
```diff
@@ -62,20 +62,20 @@ oe_providers()
 Each element in the column `database_name` is a data object that is packaged with `osmextract`. 
 You can read a detailed description of each provider data running, for example, `?geofabrik_zones` or `?bbbike_zones`. 
 
-Perhaps the best known bulk OSM data provider is Geofabrik, represented as data frame in the packaged object `geofabrik_zones`.
+Perhaps the best known bulk OSM data provider is Geofabrik, represented as `data.frame` in the packaged object `geofabrik_zones`.
```
Comment (Robinlovelace) [inline vignettes/osmextract.Rmd:65]: It's an `sf` data frame for consistency with other docs? Happy with this though.

File: vignettes/osmextract.Rmd [hunk 7]
```diff
@@ -84,77 +84,101 @@ plot(st_geometry(geofabrik_zones))
...
-plot(st_geometry(bbbike_zones), xlim = c(-2, 10), ylim = c(30, 60))
+plot(st_geometry(spData::world), xlim = c(-2, 10), ylim = c(35, 60))
+plot(st_geometry(bbbike_zones), xlim = c(-2, 10), ylim = c(35, 60), col = "darkred", add = TRUE)
 ```
 
+<img src="https://user-images.githubusercontent.com/22221146/94461461-772e4d00-01ba-11eb-950c-804ad177729f.png" width="100%" />
+
 As shown in the above visualisation of [BBBike.org](https://download.bbbike.org/osm/) zones in Europe, the provider offers rectangular extracts of major cities.
-You can also download manually selected regions of interest from the BBBike website (see also https://github.com/ITSLeeds/osmextract/issues/100).
+We are working on adding support for manually selected regions from the BBBike website (see https://github.com/ITSLeeds/osmextract/issues/100).
```
Comment (Robinlovelace) [inline vignettes/osmextract.Rmd:99]: And https://download.osmdata.xyz/ ?

File: vignettes/osmextract.Rmd [hunk 8]
```diff
@@ -217,26 +239,25 @@ You can always check the default `download_directory` used by `oe_download()` wi
...
 ## `oe_vectortranslate()`: convert to gpkg format
 
-`oe_vectortranslate()` function was defined to translate a `.pbf` file into `.gpkg` format. 
-The new `.gpkg` file is created in the same directory as the input `.pbf` file.
-The conversion is performed using [ogr2ogr](https://gdal.org/programs/ogr2ogr.html#ogr2ogr) through `vectortranslate` utility in `sf::gdal_utils()`. 
-It was created following [the suggestions](https://github.com/OSGeo/gdal/issues/2100#issuecomment-565707053) of the maintainers of GDAL.
+`oe_vectortranslate()` function translates a `.pbf` file into `.gpkg` format. 
+It takes in input a string representing the path to an existing `.pbf` file, and it returns the path to the newly generated `.gpkg` file. 
+The `.gpkg` file is created in the same directory as the input `.pbf` file and with the same name. 
+The conversion is performed using [ogr2ogr](https://gdal.org/programs/ogr2ogr.html#ogr2ogr) through `vectortranslate` utility in `sf::gdal_utils()`.
```
Comment (Robinlovelace) [inline vignettes/osmextract.Rmd:256]: :+1:

File: vignettes/providers_comparisons.Rmd [hunk 9]
```diff
@@ -0,0 +1,139 @@
+---
+title: "Comparing the supported OSM providers"
```
Comment (Robinlovelace) [inline vignettes/providers_comparisons.Rmd:2]: :+1:

**Review by Robinlovelace — APPROVED** [review]
This is a great set of improvements to the documentations, especially the vignettes. Lots of this material could go into a paper on the package at some point. Overall, big :+1:  from me.

**Comment by Robinlovelace** [comment]
There are a few questions in my comments above @agila5 which are minor, feel free to take a look at your leisure.

File: R/download.R (hunk 2 above)
Comment (agila5) [reply R/download.R:117]: Actually no since, IMO, if you are going to download a file bigger than 500MB (or some other threshold), you always want to check that the URL corresponds to the right OSM extract

File: R/match.R (hunk 4 above)
Comment (agila5) [reply R/match.R:120]: TBH I've never understood the small differences between `st_intersects`, `st_within`, `st_contain` and so on, so I decided to use `sf::st_intersects` since it is the default `op` for `[.sf`.

From [here](https://gis.stackexchange.com/questions/345184/st-intersects-vs-st-within), it looks like that there is no difference between `st_intersects` and `st_within` if you are dealing with a POINT and a POLYGON, but I agree that we should modify that part when we add support for matching using linestrings or other polygons.

File: vignettes/osmextract.Rmd (hunk 6 above)
Comment (agila5) [reply vignettes/osmextract.Rmd:65]: I decided not to change that part since the `sf` stuff is introduced a few lines below

**Comment by agila5** [comment]
> There are a few questions in my comments above @agila5 which are minor, feel free to take a look at your leisure.

Thanks for the comments

File: R/download.R (hunk 2 above)
Comment (Robinlovelace) [reply R/download.R:117]: Agreed, reasonable.

File: R/match.R (hunk 4 above)
Comment (Robinlovelace) [reply R/match.R:120]: :+1:
